src/requirements/requirement-frontend-utils.ts

```diff
@@ -524,14 +547,30 @@ export function getRelatedUnfulfilledRequirements(
           );
           if (fulfillmentStatisticsWithNewCourse.minCountFulfilled > existingMinCountFulfilled) {
             if (subRequirement.checkerWarning == null) {
-              directlyRelatedRequirements.push(subRequirement);
+              relatedRequirements.push(subRequirement);
             } else {
               selfCheckRequirements.push(subRequirement);
             }
           }
         }
+      } else {
+        selfCheckRequirements.push(subRequirement);
       }
     }
   }
-  return { directlyRelatedRequirements, selfCheckRequirements };
+  const allRequirements = [...relatedRequirements, ...selfCheckRequirements];
+  const { requirementsThatDoNotAllowDoubleCounting } = getConstraintViolationsForSingleCourse(
+    { uniqueId: -1 },
+    allRequirements.map(({ id }) => id),
+    (reqA, reqB) =>
+      allowCourseDoubleCountingBetweenRequirements(
+        userRequirementsMap[reqA],
+        userRequirementsMap[reqB]
+      )
+  );
+  const automaticallyFulfilledRequirements = relatedRequirements.filter(
+    ({ id }) => !requirementsThatDoNotAllowDoubleCounting.has(id)
+  );
```

Comment:
Everything related to `automaticallyFulfilledRequirements` fixes the following bug on staging:
College: ENG
Major: CS
Grad: None

Adding INFO 1200 to the schedule causes Engineering Communications to show up as automatically fulfilled. However, after adding the course, the requirement is not actually fulfilled. This is because the requirement is self-check, and should be excluded from that list. This bug will be irrelevant after the new add modal is implemented.

Reply:
Confirmed this is now fixed!